roles/openshift_certificate_expiry/defaults/main.yml

```diff
@@ -0,0 +1,8 @@
+---
+openshift_certificate_expiry_config_base: "/etc/origin"
```

Comment:
@jiajliu that is correct. The `openshift_certificate_expiry_config_base` key is how you specify where the base **OCP cluster configuration** directory is.

After that, the module I wrote for the role will discover the correct paths to the certificates itself. For example, if you have the following setup:

* OCP config files are installed in `/usr/share/ocp/[master/master-config.yaml, node/node-config.yaml]` 
* Your certificates are in `/usr/share/crts`
* Inside the `master/node-config.yaml` files there are keys like this:

```yaml
servingInfo:
    certFile: /path/to/name-of-cert.crt
    clientCA: /path/to/name-of-ca.crt
```

The module will read the YAML files and then parse the `certFile` and `clientCA` values out from it. If you certificates are in a special location then these two values will say where you certificates are.

In this example if we were reading `/usr/share/ocp/master/master-config.yaml` then the `[servingInfo][certFile]` value would be something like `/usr/share/crts/master.server.crt` and `node-config.yaml` could be something like `/usr/share/crts/server.crt`.

@jiajliu This role is fundamentally different from the proof of concept you saw a few months ago in that where **previously** the playbook would look for `crt` files on the file system, now we read in server config files and then read the certificates that are defined inside of the config files.

That is to say, previously we were just blindly looking for any file that ended in `.crt`, a brute-force approach. This new approach is more precise and only reads files that the service says it explicitly uses.

**Note**: Other certificates are read in different ways. For example, router/registry secrets have to be read from the `oc get secrets` command, there is no file holding the certificate data for those items.

In summary: if your cluster is alive and working then all you have to do is:

* Ensure that the default value (`/etc/origin`) for the `openshift_certificate_expiry_config_base` key is correct
* Or, ensure that you set `openshift_certificate_expiry_config_base` to point to where your actual config base is.

Locating the certificates will happen automatically.